phi/llm/knowledge/pdf.py

```diff
@@ -38,8 +38,27 @@ def get_documents(self) -> Iterator[Document]:
 
     def search(self, query: str) -> List[Document]:
         """Return all relevant documents matching the query"""
+        if self.vector_db is None:
+            logger.warning("No vector db provided")
+            return []
+
+        import openai
+
         logger.info(f"Getting relevant documents for query: {query}")
-        return []
+
+        # Get embedding for question
+        response = openai.Embedding.create(input=query, model="text-embedding-ada-002")
```

Comment:
maybe in the future, we can have an embedding class, just to avoid any openai lock-in

Reply:
@abhishek-ch added that as part of this PR